### ISSUE TITLE
Kotlin

### ISSUE THREAD

**Issue by I60R** [healarconr/logger-folding#4]
Can you add support for Kotlin and [kotlin-logging](https://github.com/MicroUtils/kotlin-logging) framework?

**Comment by healarconr** [comment]
I will give it a try.

**Comment by healarconr** [comment]
I published a 0.4.0-beta.1 version to support Kotlin and kotlin-logging. I tested it with[ the examples of the kotlin-logging wiki](https://github.com/MicroUtils/kotlin-logging/wiki#usage) and it seems to work fine. Would you help me to test it with other source code you might have?

**Comment by I60R** [comment]
Thank you,

after resolving issues with IDEA on my laptop I will add `kotlin-logging` to [android-drawable-resizer](https://github.com/I60R/android-drawable-resizer) and test on it.
Also there is one (relatively small) project at my work and probably I will have a chance to test on it

**Comment by healarconr** [comment]
@I60R Any luck with the tests? I would really like to release this feature soon.

**Comment by I60R** [comment]
I found a case, when folding don't work:
```kotlin
class HasSLF4JLogger(val logger: org.slf4j.Logger)
class Bug(override val logger: KLogger) : KLoggable {

    fun example(hasLogger: HasSLF4JLogger) = with(hasLogger) {
        this@Bug.logger.info { "don't folds" }
    }
}
```
Also I have `Alt Graph` mapped to `compose key` in my system, so default mappings don't work.
Don't sure if it's that common, but maybe better to change them to `Alt + l` and `Alt + Shift + l`?

**Comment by healarconr** [comment]
Not so long ago they changed the kotlin-logging wiki and now they recommend a different way to obtain a logger. I will test with that new way.

Is that case you found a common pattern used with kotlin-logging? I could try to make it work but I think it is highly posible to always find a case where the folding will not work if a clever or intrincate code is used because the Program Structure Interface (PSI) is different to the one supported by the plugin. I only expect to cover the most common cases and if this one is not I rather leave it as is.

Regarding the shortcut, I would prefer to continue with the current one because changing it might be an unpleasant surprise for other users and I think `Alt Gr + L` is easier to press with a single hand. In cases like yours there is always the posibility to change the shotcuts in the Settings > Keymap. There is also this new issue: #5 that recommends to support collapse by default, and with that feature a user would probably use the shorcuts less frequently.

**Comment by healarconr** [comment]
I published a new version: 0.4.0-beta.2. It should fix the folding issue of the new recommended way of using kotlin-logging and the case you found. Let me know if it works for you and if you find new cases.

**Comment by I60R** [comment]
I tested it with new way and with old - it works.  
  

Another issue found (I'm not sure if it's introduced with recent changes, so I post it here).
Steps to reproduce:

1) Ctrl-a, Fold logger method calls --> ok
2) Unfold all with Ctrl-Shift-+    --> ok
3) Ctrl-a, Fold logger method calls    --> don't fold's anything

**Comment by healarconr** [comment]
Thank you for your feedback. The new issue you mention comes from the beginning of this plugin. I was always trying to add new fold regions when the "Fold logger method calls" action was called, but if there was already a region in the same range no new region was added (which makes sense), but the existing regions remained expanded.

I published a 0.4.0-beta.3 version that addresses that issue. It should be approved and available for download in the next couple days.
